source/extensions/filters/udp/dns_filter/dns_filter.cc

```diff
@@ -0,0 +1,55 @@
+#include "extensions/filters/udp/dns_filter/dns_filter.h"
+
+#include "envoy/network/listener.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace UdpFilters {
+namespace DnsFilter {
+
+DnsFilterEnvoyConfig::DnsFilterEnvoyConfig(
+    Server::Configuration::ListenerFactoryContext& context,
+    const envoy::config::filter::udp::dns_filter::v2alpha::DnsFilterConfig& config)
+    : root_scope(context.scope()), stats_(generateStats(config.stat_prefix(), root_scope)) {
+
+  using envoy::config::filter::udp::dns_filter::v2alpha::DnsFilterConfig;
+
+  // store configured data for server context
+  const auto& server_config = config.server_config();
+
+  if (server_config.has_control_plane_cfg()) {
+
+    const auto& cfg = server_config.control_plane_cfg();
+    const size_t entries = cfg.virtual_domains().size();
+
+    virtual_domains_.reserve(entries);
+    for (const auto virtual_domain : cfg.virtual_domains()) {
+      DnsAddressList addresses{};
+
+      if (virtual_domain.endpoint().has_addresslist()) {
+        const auto& address_list = virtual_domain.endpoint().addresslist().address();
+        addresses.reserve(address_list.size());
```

Comment:
should the address here be validated its a valid ipv4 or ipv6 format?

Reply:
I'm storing the address as an InstancePointer.  When the string is parsed it's validated at that point.   I could use Address in the proto field, but that makes the config very nested.

The end result is that we end up with an exception if the address isn't a known format.  This will happen regardless of the type in the proto file.